File: leetcode2366/solution.py
class Solution:
    def minimumReplacement(self, nums: List[int]) -> int:
        ans = 0

        current = nums[-1]

        for i in range(len(nums) - 1, -1, -1):
            num = nums[i]

            if current >= num:
                current = num
                continue

            step = int(num / current) if num % current != 0 else int(num / current) - 1

            current = int(num / (step + 1))

            ans = ans + step

        return int(ans)

# class Solution:
#     def minimumReplacement(self, nums: List[int]) -> int:
#         last = nums[-1]
#         n = len(nums)
#         step = 0

#         for i in range(n-1, 0, -1):
#             if last >= nums[i - 1]:
#                 last = nums[i - 1]
#                 continue
            
#             tmp = nums[i - 1] // last

#             if nums[i - 1] % last:
#                 tmp += 1

#             last = nums[i - 1] // tmp
#             step += tmp-1

#         return step

File: leetcode2366/solution.rs
impl Solution {
    pub fn minimum_replacement(nums: Vec<i32>) -> i64 {
        // let mut length = (nums.len() - 1) as i32;
        let mut step:i64 = 0;
        let mut last:i64 = nums[nums.len() - 1] as i64;

        for num in nums.iter().rev() {
            if last >= (*num as i64) {
                last = (*num as i64);
                continue;
            }

            let mut tmp:i64 = ((*num as i64) / last) as i64;

            if ((*num as i64) % last) > 0 {
                tmp += 1;
            }

            last = ((*num as i64) / tmp) as i64;

            step += tmp-1;
            // println!("{}, {}", i, nums[i]);
            // if last >= nums[i] {
            //     last = nums[i];
            //     continue;
            // }
            // let mut tmp:i32 = (nums[i - 1] / last) as i32;

            // if (nums[i - 1] % last) > 0 {
            //     tmp += 1;
            // }

            // last = (nums[i - 1] / tmp) as i32;

            // step += tmp

        } 
        step as i64
    }
}